config/boards/radxa-e54c.csc

```diff
@@ -0,0 +1,79 @@
+# Rockchip RK3582 SoC
+BOARD_NAME="Radxa E54C"
+BOARDFAMILY="rockchip-rk3588"
+BOARD_MAINTAINER="kamilsaigol"
+BOOTCONFIG="radxa-e54c-rk3588s_defconfig"
+KERNEL_TARGET="vendor"
+BOOT_FDT_FILE="rockchip/rk3588s-radxa-e54c.dtb"
+BOOT_SCENARIO="spl-blobs"
+BOOT_SOC="rk3588"
+IMAGE_PARTITION_TABLE="gpt"
+
+
+# The kernel config hooks are always called twice, once without being in kernel directory and once with current directory being the kernel work directory.
+# Enable the distributed switch architecture to expose the individual LAN and WAN interfaces.
+function custom_kernel_config__radxa_e54c_enable_ethernet_switch_arch() {
+	if [[ ! -f .config ]]; then
+		# kernel_config_modifying_hashes is only needed during first call of this function to calculate kernel artifact version string. It serves no purpose whatsoever during second call.
+		kernel_config_modifying_hashes+=("CONFIG_NET_DSA=m" "CONFIG_NET_DSA_REALTEK=m" "CONFIG_NET_DSA_REALTEK_MDIO=m" "CONFIG_NET_DSA_REALTEK_SMI=m" "CONFIG_NET_DSA_REALTEK_RTL8365MB=m" "CONFIG_NET_DSA_REALTEK_RTL8366RB=m")
+	else
+		display_alert "$BOARD" "Enable Realtek Distributed Switch" "info"
+		kernel_config_set_m NET_DSA
+		kernel_config_set_m NET_DSA_REALTEK
+		kernel_config_set_m NET_DSA_REALTEK_MDIO
+        kernel_config_set_m NET_DSA_REALTEK_SMI
+        kernel_config_set_m NET_DSA_REALTEK_RTL8365MB
+        kernel_config_set_m NET_DSA_REALTEK_RTL8366RB
+	fi
+}
```

Comment:
@kamilsaigol Sorry for being late, but this doesn't belong here.
Boards shouldn't modify kernel .config, which is associated with the `LINUXFAMILY`(/`BOARDFAMILY`), not `BOARD`.
Please move this to `.config` where it belongs -- otherwise kernel packages published to apt repo will be inconsistent (images would work fine though.)

Reply:
Damn, i missed this too. Here too: https://github.com/armbian/build/pull/8665/files#diff-be8acfa77c3b0659fe481d1833cf9e705de3d7652d15678e6e14c216ff5cee31

Reply:
@coderabbitai alert people to not use kernel_config_set in the board config. Boards have common kernel.